App/Talent.App.WebApp/wwwroot/js/react/ReactScripts/TalentFeed/TalentCard.jsx

```diff
@@ -1,18 +1,120 @@
+                    {/*<source src="mov_bbb.mp4" type="video/mp4" />*/}
+                    {/*<source src="mov_bbb.ogg" type="video/ogg" />*/}
+                    <source src="mov.mp4" type="video/mp4" />
+                    <source src="mov.ogg" type="video/ogg" />
+                    Your browser does not support HTML5 video.
+                </video>
+            </Card.Content>
+        )
+    }
+
+    snapshotButton() {
+        return (
+            <Grid.Column className="center aligned">
+                <Icon size="large"
+                    name="user"
+                    link
+                    onClick={() => this.setState({ showSnapshot: true })}
+                />
+            </Grid.Column>
+        )
+    }
+
+    videoButton() {
+        return (
+            <Grid.Column className="center aligned">
+                <Icon size="large"
+                    name="video camera"
+                    link
+                    onClick={() => this.setState({ showSnapshot: false })}
+                />
+            </Grid.Column>
+        )
+    }
+
     render() {
-       
+        if (this.props.feed) {
+            const { showSnapshot } = this.state;
+            const { name, skills } = this.props.feed;
+
+            return (
+                <Card fluid>
+                    <Card.Content>
+                        <Card.Header>{name}
+                            <div className="ui right floated">
+                                <Icon name="star" size="big" />
+                            </div>
+                        </Card.Header>
+                    </Card.Content>
+                    {showSnapshot ? this.renderSnapshot() : this.renderVideo()}
+                    <Card.Content >
+                        <Grid columns={4}>
+                            {showSnapshot ? this.videoButton() : this.snapshotButton()}
+                            <Grid.Column className="center aligned">
+                                <Icon size="large" name="file pdf outline" link />
+                            </Grid.Column>
+                            <Grid.Column className="center aligned">
+                                <Icon size="large" name="linkedin" link />
+                            </Grid.Column>
+                            <Grid.Column className="center aligned">
+                                <Icon size="large" name="github" link />
+                            </Grid.Column>
+                        </Grid>
+                    </Card.Content>
+                    <Card.Content >
+                        {skills.length > 0 ?
+                            skills.map((skill, index) =>
+                                <Button key={index} basic size="tiny" compact color="blue" content={skill} />
+                            )
+                            :
+                            <Button basic disabled size="tiny" compact color="grey" content="N/A" />
+                        }
```

Comment:
This is fine, but you can improve this by creating a new function and render the elements in there. This is because if you get used to this, it will not be a good practice implementing multiple conditions in the render function and it will be hard for you to manage it in the long run.

Reply:
Created new function renderSkills() to call from render function.